framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1100,13 +1098,20 @@ public AnnotatedTypeMirror fromExpression(ExpressionTree tree) {
     }
 
     /**
-     * Determines the annotated type from a type in tree form.  This method
-     * does not add implicit annotations.
+     * Creates an AnnotatedTypeMirror for the tree.  The AnnotatedTypeMirror contains
+     * annotations explicitly written on the tree and annotations inherited from the class
+     * declaration of types {@link #annotateInheritedFromClass(AnnotatedTypeMirror)}.
+     * It also adds type arguments to raw types that include annotations from the element
+     * declaration of the type {@link #fromElement(Element)}.
+     * <p>
+     * Called on the following trees: AnnotatedTypeTree, ArrayTypeTree, ParameterizedTypeTree,
+     * PrimitiveTypeTree, TypeParameterTree, WildcardTree, UnionType,
+     * IntersectionTypeTree, and IdentifierTree, MemberSelectTree.
      *
      * @param tree the type tree
-     * @return the annotated type of the type in the AST
+     * @return the (partially) annotated type of the type in the AST
```

Comment:
Can whether the result is partially or fully annotated be made explicit in the method name?  It seems that it would be hard to keep straight what each method does.